docs/madr/decisions/046-policy-on-namespace.md

```diff
@@ -0,0 +1,456 @@
+The label is part of the hash suffix when policy is synced to global.
+
+Policy applied to the custom namespace `my-namespace` of zone-a, won’t affect pods in the custom
+namespace `my-namespace` of zone-b,
+as these namespaces are considered different.
+If such behavior is need, user can always apply the policy on global with top-level targetRef:
+
+```yaml
+targetRef:
+  kind: MeshSubset
+  tags:
+    k8s.kuma.io/namespace: my-namespace
+```
+
+### Order when merging
+
+By design, several policies can select the same DPP.
+In that case, policies are ordered and their `to` (or `from`) lists are merged.
+Introducing namespace-scoped policies adds another step to the policy comparison process:
+
+1. Is top-level `targetRef.kind` more specific (MeshServiceSubset > MeshService > MeshSubset > Mesh)?
+   When kinds are equal, go to the next step.
+2. Zone originated policy is more specific than global originated policy.
+   When policies originates at the same place, go to the next step.
+3. **[new step]** Policy in a custom namespace is more specific than a policy in the `kuma-system` namespace.
+   When policies applied in the same namespace, go to the next step.
+4. The policy is more specific if it's name is lexicographically less than other policy's name ("aaa" < "bbb" so that "
+   aaa" policy is more specific)
+
+### Cross-policy references
+
+Kuma policies support cross-policy references.
+At this moment, it works only between MeshTimeout and MeshHTTPRoute, but there are plans to support it for other
+policies, i.e. [#6645](https://github.com/kumahq/kuma/issues/6645).
+Since policies will be namespace-scoped, we need to add a new field `Namespace` in the TargetRef:
+
+```go
+// TargetRef defines structure that allows attaching policy to various objects
+type TargetRef struct {
+    // Kind of the referenced resource
+    // +kubebuilder:validation:Enum=Mesh;MeshSubset;MeshGateway;MeshService;MeshServiceSubset;MeshHTTPRoute
+    Kind TargetRefKind `json:"kind,omitempty"`
+    // Name of the referenced resource. Can only be used with kinds: `MeshService`,
+    // `MeshServiceSubset` and `MeshGatewayRoute`
+    Name string `json:"name,omitempty"`
+    // Tags used to select a subset of proxies by tags. Can only be used with kinds
+    // `MeshSubset` and `MeshServiceSubset`
+    Tags map[string]string `json:"tags,omitempty"`
+    // Mesh is reserved for future use to identify cross mesh resources.
+    Mesh string `json:"mesh,omitempty"`
+    // ProxyTypes specifies the data plane types that are subject to the policy. When not specified,
+    // all data plane types are targeted by the policy.
+    // +kubebuilder:validation:MinItems=1
+    ProxyTypes []TargetRefProxyType `json:"proxyTypes,omitempty"`
+    // Namespace of the referenced resource. If not specified, then equal the value is equal to the resource's namespace
+    // where this TargetRef is used.
+    Namespace string `json:"namespace,omitempty"`
```

Comment:
It wasn't possible to create policies on the namespace before. So I guess we are not constrained by the backward compatibility here

Reply:
So the comment is a little weird. Because there are currently policies and they have empty namespace and live in the system namespace. This means that they'd match things in the system namespace no?

Reply:
btw there is a typo in the comment, should be without "equal":
> Namespace of the referenced resource. If not specified, then the value is equal to the resource's namespace where this TargetRef is used.

If today you have a MeshTimeout that references MeshHTTPRoute with an empty namespace like:
```yaml
kind: MeshTimeout
metadata:
  name: timeout-1
  namespace: kuma-system
spec:
  targetRef:
    kind: MeshHTTPRoute
    name: route-1
```
then Kuma searches `route-1` in `kuma-system`. And after the upgrade to 2.8 it won't change, as the default value for `spec.targetRef.namespace` is `metadata.namespace`.

Reply:
Yes but what about:

```yaml
kind: MeshTimeout
metadata:
  name: timeout-1
  namespace: kuma-system
spec:
  targetRef:
    kind: Mesh
```
?

This should match all dpps regardless of location no?

Reply:
A Mesh is cluster scoped. The only case we have to worry about not breaking is `namespace: kuma-system` `refers to` `namespace: ""` and either the referred to object is cluster scoped, in which case namespace never makes sense, or it was namespace scoped, in which case it must have been in `kuma-system` so nothing changes.